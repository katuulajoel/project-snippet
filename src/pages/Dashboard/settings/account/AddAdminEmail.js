/* eslint-disable react/no-unescaped-entities */
/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React from "react";
import PropTypes from "prop-types";
import { Row, Col } from "reactstrap";

/* ------------------------- Component dependencies ------------------------- */
import { AnimatedButton } from "../../../../components/Button";
import FormInput from "./FormInput";

const AddAdminEmail = ({ onSave }) => {
  return (
    <form onSubmit={onSave} aria-label="addAdminEmail-form" id="addAdminEmail">
      <div className="header">Add admin email address for invoices</div>
      <FormInput label="New Email Address" name="invoice_email" type="email" />
      <Row>
        <Col>
          <AnimatedButton
            aria-label="addAdminEmail-submit"
            targetform="#addAdminEmail"
            className="loader-opaque"
          >
            Save
          </AnimatedButton>
        </Col>
      </Row>
    </form>
  );
};

AddAdminEmail.propTypes = {
  onSave: PropTypes.func,
};

export default AddAdminEmail;
