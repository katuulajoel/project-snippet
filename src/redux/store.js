import { applyMiddleware, createStore } from "redux";
import reduxThunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

import monitorReducersEnhancer from "../configs/enhancers/monitorReducer";
import loggerMiddleware from "../configs/middleware/logger";
import rootReducer from "./reducers";

export function configureStore(preloadedState = {}) {
  const middlewares = [loggerMiddleware, reduxThunk];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer, monitorReducersEnhancer];
  const composedEnhancers = composeWithDevTools(...enhancers);

  const store = createStore(rootReducer, preloadedState, composedEnhancers);

  if (process.env.NODE_ENV !== "production" && module.hot) {
    module.hot.accept("./reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
}

const store = configureStore();
export default store;
