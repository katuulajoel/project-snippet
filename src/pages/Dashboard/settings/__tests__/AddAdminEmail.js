import React from "react";
import { Provider } from "react-redux";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import AddAdminEmail from "../account/AddAdminEmail";
import store from "../../../../redux/store";

afterEach(cleanup);
describe("AddAdminEmail layout test", () => {
  it("Snapshot test for AddAdminEmail component", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <AddAdminEmail />
        </Router>
      </Provider>
    );

    expect(asFragment(<AddAdminEmail />)).toMatchSnapshot();
  });
});
