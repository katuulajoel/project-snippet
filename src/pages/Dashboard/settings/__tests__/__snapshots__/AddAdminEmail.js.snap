// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAdminEmail layout test Snapshot test for AddAdminEmail component 1`] = `
<DocumentFragment>
  <form
    aria-label="addAdminEmail-form"
    id="addAdminEmail"
  >
    <div
      class="header"
    >
      Add admin email address for invoices
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
      >
        New Email Address
        <span
          class="label-style"
        >
          *
        </span>
      </label>
      <div
        class="sc-bdnxRM lboORq"
        data-testid="input"
      >
        <input
          aria-label="invoice_email"
          class="form-control"
          data-testid=""
          id=""
          label="Label"
          name="invoice_email"
          placeholder="Enter your password"
          required=""
          type="email"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <button
          class="btn btn-default btn btn-primary save  "
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
